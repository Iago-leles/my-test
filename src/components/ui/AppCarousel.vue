<template>
  <div class="relative-position">
    <q-carousel
      v-model="slide"
      control-color="white"
      control-text-color="black"
      control-type="regular"
      arrows
      infinite
      :autoplay="autoplay"
      height="300px"
      class="carousel-card"
      :style="fullRounded ? 'border-radius: 14px !important' : ''"
    >
      <q-carousel-slide v-for="(img, index) in images" :key="index" :name="index" class="q-pa-none">
        <q-img :src="img" style="height: 100%" />
      </q-carousel-slide>
    </q-carousel>

    <div v-if="showCounter" class="absolute bg-grey-10 text-white q-px-md counter">
      {{ `${slide + 1} / ${images.length}` }}
    </div>
  </div>
</template>

<script setup lang="ts">
import { ref } from 'vue'

defineProps<{
  images: string[]
  autoplay?: boolean
  showCounter?: boolean
  fullRounded?: boolean
}>()

const slide = ref(0)
</script>

<style scoped lang="scss">
.carousel-card {
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

.counter {
  bottom: 14px;
  left: 50%;
  transform: translateX(-50%);
  opacity: 0.8;
  border-radius: 10px;
}
</style>
