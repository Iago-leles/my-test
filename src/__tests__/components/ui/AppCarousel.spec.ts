import { mount } from '@vue/test-utils'
import { expect, describe, it } from 'vitest'
import AppCarousel from '@/components/ui/AppCarousel.vue'

describe('AppCarousel', () => {
  const images = ['image1.jpg', 'image2.jpg', 'image3.jpg']

  it('should match snapshot', () => {
    const wrapper = mount(AppCarousel, {
      props: { images },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should show counter when showCounter prop is true', () => {
    const wrapper = mount(AppCarousel, {
      props: {
        images,
        showCounter: true,
      },
    })

    expect(wrapper.find('.counter').exists()).toBe(true)
    expect(wrapper.find('.counter').text()).toBe('1 / 3')
  })

  it('should not show counter when showCounter prop is false', () => {
    const wrapper = mount(AppCarousel, {
      props: {
        images,
        showCounter: false,
      },
    })

    expect(wrapper.find('.counter').exists()).toBe(false)
  })

  it('should apply full rounded style when fullRounded prop is true', () => {
    const wrapper = mount(AppCarousel, {
      props: {
        images,
        fullRounded: true,
      },
    })

    expect(wrapper.find('.carousel-card').attributes('style')).toContain('border-radius: 14px')
  })
})
