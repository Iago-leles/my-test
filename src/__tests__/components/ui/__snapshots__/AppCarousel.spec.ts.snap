// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppCarousel > should match snapshot 1`] = `
"<div data-v-813af4b2="" class="relative-position">
  <div data-v-813af4b2="" class="q-carousel q-panel-parent q-carousel--without-padding q-carousel--arrows-horizontal carousel-card" style="height: 300px;">
    <div class="q-carousel__slides-container">
      <div class="q-panel scroll" style="--q-transition-duration: 300ms;" role="tabpanel">
        <div data-v-813af4b2="" class="q-carousel__slide q-pa-none">
          <div data-v-813af4b2="" class="q-img q-img--menu" style="height: 100%;" role="img">
            <div style="padding-bottom: 56.24929688378895%;"></div>
            <div class="q-img__container absolute-full"><img class="q-img__image q-img__image--with-transition q-img__image--current" style="object-fit: cover; object-position: 50% 50%;" loading="lazy" fetchpriority="auto" aria-hidden="true" draggable="false" src="image1.jpg"></div>
            <transition-stub name="q-transition--fade" appear="false" persisted="false" css="true">
              <div class="q-img__loading absolute-full flex flex-center"><svg class="q-spinner q-spinner-mat" width="1em" height="1em" viewBox="25 25 50 50">
                  <circle class="path" cx="50" cy="50" r="20" fill="none" stroke="currentColor" stroke-width="5" stroke-miterlimit="10"></circle>
                </svg></div>
            </transition-stub>
          </div>
        </div>
      </div>
    </div>
    <div class="q-carousel__control q-carousel__arrow q-carousel__prev-arrow q-carousel__prev-arrow--horizontal absolute flex flex-center"><button class="q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--round bg-white text-black q-btn--actionable q-focusable q-hoverable q-btn--dense" tabindex="0" type="button" regular="true"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row"><i class="q-icon notranslate material-icons" aria-hidden="true" role="img">chevron_left</i></span></button></div>
    <div class="q-carousel__control q-carousel__arrow q-carousel__next-arrow q-carousel__next-arrow--horizontal absolute flex flex-center"><button class="q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--round bg-white text-black q-btn--actionable q-focusable q-hoverable q-btn--dense" tabindex="0" type="button" regular="true"><span class="q-focus-helper"></span><span class="q-btn__content text-center col items-center q-anchor--skip justify-center row"><i class="q-icon notranslate material-icons" aria-hidden="true" role="img">chevron_right</i></span></button></div>
  </div>
  <!--v-if-->
</div>"
`;
